python/rascaline/examples/le-basis.py

```diff
@@ -0,0 +1,203 @@
+
+keys = []
+blocks = []
+for center_species in all_species:
+    for neighbor_species in all_species:
+        for l in range(l_max+1):
+            keys.append([l, center_species, neighbor_species])
+            blocks.append(
+                TensorBlock(
+                    values=np.empty((1, n_max_l[l])),
+                    samples=Labels.single(),
+                    components=[],
+                    properties=Labels(
+                        names=["n"],
+                        values=np.arange(n_max_l[l]).reshape(n_max_l[l], 1)
+                    )
+                )
+            )
+
+selected_properties = TensorMap(
+    keys=Labels(
+        names=["spherical_harmonics_l", "species_center", "species_neighbor"],
+        values=np.array(keys)
+    ),
+    blocks=blocks
+)
+
+# %%
+#
+# Build a calculator and calculate the spherical expansion
+
+spliner = rascaline.utils.SoapSpliner(
+    cutoff=cutoff,
+    max_radial=n_max,
+    max_angular=l_max,
+    basis=rascaline.utils.SphericalBesselBasis(cutoff=cutoff, max_radial=n_max, max_angular=l_max),
+    density=rascaline.utils.DeltaDensity(),
+    accuracy=1e-8,
+)
+
+calculator = rascaline.SphericalExpansion(
+    cutoff=cutoff,
+    max_radial=n_max,
+    max_angular=l_max,
+    center_atom_weight=1.0,
+    radial_basis=spliner.compute(),
+    atomic_gaussian_width=-1.0,  # will not be used due to the delta density above
+    cutoff_function={"Step": {}},
+)
+
+spherical_expansion = calculator.compute(
+    structures,
+    selected_properties=selected_properties  # we tell the calculator to only compute the selected properties
+)
+
+if HAS_TORCH_SPEX:
+    # check that the results are the same as those from torch_spex
+    from torch_spex.spherical_expansions import SphericalExpansion
+    from torch_spex.structures import InMemoryDataset, TransformerNeighborList, collate_nl
+    hypers = {
+        "cutoff radius": cutoff,
+        "radial basis": {
+            "type": "le",
+            "mlp": False,
+            "r_cut": cutoff,
+            "E_max": E_max,
+            "normalize": False
+        }
+    }
+
+    transformers = [TransformerNeighborList(cutoff=cutoff, device="cpu")]
+    dataset = InMemoryDataset(structures, transformers)
+    dataloader = torch.utils.data.DataLoader(dataset, batch_size=10, shuffle=False, collate_fn=collate_nl)
+    structures = next(iter(dataloader))
+
+    torch_spex_calculator = SphericalExpansion(hypers, all_species, device="cpu")
+    torch_spex_spherical_expansion = torch_spex_calculator(**structures)
+    spherical_expansion = spherical_expansion.keys_to_properties(["species_neighbor"])
+
+    block = spherical_expansion.block(spherical_harmonics_l=1, species_center=1)
+    torch_spex_block = torch_spex_spherical_expansion.block({"lam":1, "a_i":1, "sigma":1})
+
+    assert np.allclose(block.values[:, :, 1], torch_spex_block.values.numpy()[:, :, 1])
```

Comment:
This should not be in the example, it looks more like a test. I'm happy to have this as a test of the splining+properties selection code, but this will just confuse users trying to read the example.

Reply:
We can't have tests depend on torch_spex here at the moment, I would rather add the corresponding tests to torch_spex

Reply:
as you prefer, if torch_spex is relatively stable we could also add it to `tox.ini`!

Reply:
I've opened an issue in torch_spex, I'll try to do it relatively soon